MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -413,6 +421,9 @@ protected virtual void Dispose(bool disposing)
                     // Dispose of all remaining graphics resources before disposing of the graphics device
                     GraphicsResource.DisposeAll();
 
+                    // Clear the effect cache.
+                    EffectCache.Clear();

```

Comment:
Yes - I stepped through in the debugger, and the `Effect` I was testing with was disposed in `GraphicsResource.DisposeAll()`.